iceoryx_examples/automotive_soa/iox_automotive_proxy.cpp

```diff
@@ -0,0 +1,188 @@
+        };
+
+        //! [set up asynchronous search]
+        auto handle = MinimalProxy::EnableFindServiceCallback(callback, exampleInstanceSearchQuery);
+        //! [set up asynchronous search]
+        maybeHandle.emplace(handle);
+        std::cout << "  Waiting for instance called '" << exampleInstanceSearchQuery.c_str()
+                  << "' to become available.." << std::endl;
+    }
+    //! [Field: create ints for computeSum]
+    uint64_t addend1{0};
+    uint64_t addend2{0};
+    //! [Field: create ints for computeSum]
+
+    while (!iox::posix::hasTerminationRequested())
+    {
+        {
+            //! [gain exclusive access to proxy]
+            auto proxyGuard = maybeProxy.getScopeGuard();
+            if (proxyGuard->has_value())
+            {
+                auto& proxy = proxyGuard->value();
+                //! [gain exclusive access to proxy]
+
+                //! [Event: receiveCallback]
+                auto onReceive = [&]() -> void {
+                    proxy.m_event.TakeNewSamples([](const auto& topic) {
+                        auto finish = std::chrono::steady_clock::now();
+                        std::cout << "Event: value is " << topic->counter << std::endl;
+                        auto duration =
+                            std::chrono::duration_cast<std::chrono::nanoseconds>(finish - topic->sendTimestamp);
+                        std::cout << "Event: latency (ns) is " << duration.count() << std::endl;
+                    });
+                };
+                //! [Event: receiveCallback]
+
+                //! [Event: set receiveCallback]
+                if (!proxy.m_event.HasReceiveCallback())
+                {
+                    proxy.m_event.Subscribe(10U);
+                    proxy.m_event.SetReceiveCallback(onReceive);
+                }
+                //! [Event: set receiveCallback]
+
+                //! [Field: get and set data]
+                auto fieldFuture = proxy.m_field.Get();
+                try
+                {
+                    auto result = fieldFuture.get();
+                    std::cout << "Field: value is " << result.counter << std::endl;
+
+                    if (result.counter >= 4242)
+                    {
+                        result.counter++;
+                        proxy.m_field.Set(result);
+                        std::cout << "Field: value set to " << result.counter << std::endl;
+                    }
+                }
+                catch (const std::future_error&)
+                {
+                    std::cout << "Empty future from field received, please start the 'iox-cpp-automotive-skeleton'."
+                              << std::endl;
+                }
+                //! [Field: get and set data]
+
+                //! [Method: call computeSum remotely]
+                auto methodFuture = proxy.computeSum(addend1, addend2);
+                try
+                {
+                    auto result = methodFuture.get();
```

Comment:
Call `methodFuture.valid()` first and then `get` to avoid the exception and this exception handling.